File: test-plugins/org.yakindu.sct.generator.csharp.test/test-gen/AssignmentAsExpressionTest.cs
/**
* Copyright (c) 2016 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
/**
 *  Unit TestCase for AssignmentAsExpression
 */
using System;
namespace Yakindu.SCR.AssignmentAsExpression {
	using NUnit.Framework;
	using Yakindu.SCR;
	
	[TestFixture]
	public class AssignmentAsExpressionTest {
		private AssignmentAsExpressionStatemachine sct;
	
	[SetUp]
	public void setup() {
		sct = new AssignmentAsExpressionStatemachine();
		sct.init();
	}
	
		[Test]
		public void testsimpleAssignment(){
			sct.enter();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_Add));
			Assert.IsTrue(sct.getB() == 5L);
			Assert.IsTrue(sct.getA() == 9L);
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_Subtract));
			Assert.IsTrue(sct.getD() == 6L);
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_Multiply));
			Assert.IsTrue(sct.getE() == 15L);
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_Divide));
			Assert.IsTrue(sct.getG() == 1L);
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_Modulo));
			Assert.IsTrue(sct.getI() == 1L);
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_Shift));
			Assert.IsTrue(sct.getJ() == 16L);
			Assert.IsTrue(sct.getK() == 4L);
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_boolean_And));
			Assert.IsTrue(sct.getL() == 1L);
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_boolean_Or));
			Assert.IsTrue(sct.getP() == 15L);
			sct.runCycle();
			Assert.IsTrue(sct.isStateActive(AssignmentAsExpressionStatemachine.State.main_region_boolean_Xor));
			Assert.IsTrue(sct.getU() == 12L);
			sct.exit();
		}
	}
}

File: test-plugins/org.yakindu.sct.generator.cpp.test/gtests/AssignmentAsExpressionTest/AssignmentAsExpressionTest.cc
#include <string>
#include "gtest/gtest.h"
#include "AssignmentAsExpression.h"
#include "sc_runner.h"
#include "sc_types.h"

namespace  {

AssignmentAsExpression* statechart;



//! The timers are managed by a timer service. */
static SctUnitRunner * runner;

class AssignmentAsExpressionTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		statechart = new AssignmentAsExpression();
		statechart->init();
		runner = new SctUnitRunner(
			statechart,
			false,
			200
		);
	}
	virtual void TearDown() {
		delete statechart;
		delete runner;
	}
};


TEST_F(AssignmentAsExpressionTest, simpleAssignment) {
	
	statechart->enter();
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_Add));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_b()== 5);
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_a()== 9);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_Subtract));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_d()== 6);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_Multiply));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_e()== 15);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_Divide));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_g()== 1);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_Modulo));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_i()== 1);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_Shift));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_j()== 16);
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_k()== 4);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_boolean_And));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_l()== 1);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_boolean_Or));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_p()== 15);
	
	runner->proceed_cycles(1);
	
	EXPECT_TRUE(statechart->isStateActive(AssignmentAsExpression::main_region_boolean_Xor));
	
	EXPECT_TRUE(statechart->getDefaultSCI()->get_u()== 12);
	
	statechart->exit();
	
	
}

}

